src/backend/distributed/commands/alter_table.c

```diff
@@ -728,7 +728,7 @@ ConvertTable(TableConversionState *con)
 		Node *parseTree = ParseTreeNode(attachPartitionCommand);
 
 		ProcessUtilityParseTree(parseTree, attachPartitionCommand,
-								PROCESS_UTILITY_TOPLEVEL,
+								PROCESS_UTILITY_QUERY,
 								NULL, None_Receiver, NULL);
```

Comment:
(unrelated) a bit surprising that we do ProcessUtilityParseTree here, but ExecuteQueryViaSPI above.